pkg/db/src/implementations/version-1.1/note.ts

```diff
@@ -211,51 +211,52 @@ class Note implements NoteInterface {
 	}
 
 	/**
-	 * Add volumes accepts a tuple update (isbn, quantity, warehouse) or an array of tuples (updates) for
+	 * Add volumes accepts an object or multiple objects of VolumeStock updates (isbn, quantity, warehouseId?) for
 	 * book quantities. If a volume with a given isbn is found, the quantity is aggregated, otherwise a new
 	 * entry is pushed to the list of entries.
 	 */
 	addVolumes(...params: Parameters<NoteInterface['addVolumes']>): Promise<NoteInterface> {
-		// A check to see if the update function should be ran once or multiple times
-		const isTuple = (params: Parameters<NoteInterface['addVolumes']>): params is VolumeTransactionTuple => {
-			return !Array.isArray(params[0]);
-		};
-
 		return runAfterCondition(async () => {
-			const updateQuantity = (isbn: string, quantity: number, warehouseId = this.#w._id) => {
-				const matchIndex = this.entries.findIndex((entry) => entry.isbn === isbn && entry.warehouseId === warehouseId);
+			params.forEach((update) => {
+				const warehouseId = this.noteType === 'inbound' ? this.#w._id : update.warehouseId ? versionId(update.warehouseId) : '';
+				const matchIndex = this.entries.findIndex(
+					(entry) => entry.isbn === update.isbn && entry.warehouseId === update.warehouseId
+				);
 
 				if (matchIndex === -1) {
-					this.entries.push({ isbn, warehouseId, quantity });
+					this.entries.push({ isbn: update.isbn, warehouseId, quantity: update.quantity });
 					return;
 				}
 
 				this.entries[matchIndex] = {
-					isbn,
+					isbn: update.isbn,
 					warehouseId,
-					quantity: this.entries[matchIndex].quantity + quantity
+					quantity: this.entries[matchIndex].quantity + update.quantity
 				};
-			};
-
-			if (isTuple(params)) {
-				updateQuantity(params[0], params[1], params[2]);
-			} else {
-				params.forEach((update) => updateQuantity(update[0], update[1], update[2]));
-			}
+			});
 
 			return this.update(this, {});
 		}, this.#initialized);
 	}
 
-	updateTransaction(transaction: VolumeStock): Promise<NoteInterface> {
+	updateTransaction(transaction: PickPartial<VolumeStock, 'warehouseId'>): Promise<NoteInterface> {
 		// Create a safe copy of volume entries
 		const entries = [...this.entries];
-		const i = entries.findIndex(({ isbn }) => isbn === transaction.isbn);
+
+		const versionedTransaction = { ...transaction, warehouseId: transaction.warehouseId ? versionId(transaction.warehouseId) : '' };
+		// handle edge case where we have multiple books with the same isbn, but belonging to different warehouses
+		// match isbn only in case entry has no whId but transaction does (not undefined or empty string)
+		const i = entries.findIndex(
+			({ isbn, warehouseId }) =>
+				isbn === versionedTransaction.isbn &&
+				(warehouseId === versionedTransaction.warehouseId || (!warehouseId && transaction.warehouseId))
```

Comment:
This is not handling all the warehouse update cases...I'll remove it (simplify so that it only performs updates requested in this ticket) and we should think about how to handle warehouse updates.

But it's a valid point and a functionality we'll definitely need.